src/docs/testing/integration-tests/index.md

```diff
@@ -0,0 +1,290 @@
+
+### Test Lab Project Setup
+
+Go to the [Firebase Console][], and create a new project if you don't have one
+already. Then navigate to Quality > Test Lab:
+
+{% asset integration-test/test-lab-1.png class="mw-100" alt="Firebase Test Lab Console" %}
+
+### Uploading an Android APK
+
+Create an APK using Gradle:
+
+```bash
+pushd android
+# flutter build generates files in android/ for building the app
+flutter build apk
+./gradlew app:assembleAndroidTest
+./gradlew app:assembleDebug -Ptarget=integration_test/<name>_test.dart
+popd
+```
+
+Where `<name>_test.dart` is the file created in the Project Setup section.
+
+Drag the "debug" APK from
+`<flutter_project_directory>/build/app/outputs/apk/debug` into the "Android Robo
+Test" target on the web page. This will start a Robo test and allow you to run
+other tests:
+
+{% asset integration-test/test-lab-2.png class="mw-100" alt="Firebase Test Lab upload" %}
+
+Click **Run a test**, select the **Instrumentation** test type and drag these
+two files:
+ sdfasdf asdf 
+ 
+ * `<flutter_project_directory>/build/app/outputs/apk/debug/<file>.apk`
+ * `<flutter_project_directory>/build/app/outputs/apk/androidTest/debug/<file>.apk`
+
+{% asset integration-test/test-lab-3.png class="mw-100" alt="Firebase Test Lab upload two APKs" %}
+
+If a failure occurs, you can view the output by selecting the red icon:
+
+{% asset integration-test/test-lab-4.png class="mw-100" alt="Firebase Test Lab test results" %}
+
+### Uploading an Android APK from the command line
+
+See the [Firebase Test Lab section of the README][] for instructions on uploading
+the APKs from the command line.
+
+## Migrating from flutter_driver
+
+To migrate from flutter_driver, follow these steps:
+ 
+1. Remove any calls to `enableFlutterDriverExtension()` in your
+  application. 
+2. Migrate any `test_driver` scripts to use `package:integration_test` and
+  WidgetTester instead of package:flutter_driver.
+
+Before:
+
+<?code-excerpt "integration_test/test_driver/before.dart"?>
+```dart
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('end-to-end test', () {
+    FlutterDriver driver;
+
+    setUpAll(() async {
+      // Connect to a running Flutter application instance.
+      driver = await FlutterDriver.connect();
+    });
+
+    tearDownAll(() async {
+      if (driver != null) driver.close();
+    });
+
+    test('tap on the floating action button; verify counter', () async {
+      // Finds the floating action button to tap on.
+      SerializableFinder fab = find.byTooltip('Increment');
+
+      // Wait for the floating action button to appear.
+      await driver.waitFor(fab);
+
+      // Emulate a tap on the floating action button.
+      await driver.tap(fab);
+
+      // Wait for text to change to the desired value.
+      await driver.waitFor(find.text('1'));
+    });
+  });
+}
+```
+
+After:
```

Comment:
Put the "after" code into its own code block. Introduce with "Code after migration".

Reply:
It already is in it's own block? I changed this to "Code after migration".